src/fixate/drivers/handlers.py

```diff
@@ -0,0 +1,76 @@
+"""
+This module implements concrete AddressHandler type, that
+can be used to implement IO for the fixate.core.switching module.
+"""
+from __future__ import annotations
+
+from typing import Sequence
+import itertools
+
+from fixate.core.switching import Pin, PinValueAddressHandler
+from fixate.drivers import ftdi
+
+
+def _pins_for_one_relay_matrix(relay_matrix_num: int) -> list[Pin]:
+    """
+    A helper to create pin names for relay matrix cards.
+
+    Returns 16 pin names. If relay_matrix_num is 1:
+    1K1, 1K2, 1K3, ..., 1K16
+    """
+    return [f"{relay_matrix_num}K{relay}" for relay in range(1, 17)]
+
+
+# This is a real quick test. Not worth the effort unravelling
+# imports when ftdi isn't importable, just to get this into a
+# proper test right now...
+__expected = (
+    "3K1 3K2 3K3 3K4 3K5 3K6 3K7 3K8 3K9 3K10 3K11 3K12 3K13 3K14 3K15 3K16".split()
+)
+assert _pins_for_one_relay_matrix(3) == __expected
+
+
+class RelayMatrixAddressHandler(PinValueAddressHandler):
+    """
+    An address handler which uses the ftdi driver to control pins.
+
+    We create this concrete address handler because we use it most
+    often. FT232 is used to bit-bang to shift register that are control
+    the switching in a jig.
+    """
+
+    def __init__(
+        self,
+        ftdi_description: str,
+        relay_matrix_count: int,
+        extra_pins: Sequence[Pin] = tuple(),
+    ) -> None:
+        relay_matrix_pin_list = tuple(
```

Comment:
There is a jig that using the follow pins, which match up with schematic designators: RC1K1-16, RC2K1-16, RA1K1-16, RA2K1-16, RA3K1-16, RP1K1-16, RP2K1-16, RH1K1-16

So I'm wondering if we might need to make this a bit more flexible. Although, even as is you could pass in 0 for `relay_matrix_count` and build a pin list anyway you want to pass into `extra_pins`

Perhaps all we need to do is make `_pins_for_one_relay_matrix` a public function and change it to accept a prefix rather than an integer for the designator.

Reply:
I like the suggestion to make `_pins_for_one_relay_matrix` public and accepting a prefix. I'm assuming the RA, RC etc have meaning for the specific jig, and there may be other instances where this sort of thing would help identify what matrix is what and I like to have jig schematic and script designations for relays in parity.

Reply:
Yeah, those different designators indicate schematic sheet and different PCBs controlled from the one ftdi.

I'll collect a few example and see if I can come up with a simple enough API that fit our current diversity of usage

Reply:
I've changed this around a bit. I've made the handler so that you have to pass in an pin list. Then I've created to helper functions that make it easy to build pin lists similar to the ones I could find in existing scripts.

One potential sticking point - I've not provided the options to use underscores. For example you can only do pins like `RM3K1` not `RM3_K1`. I'm keen to avoid trying to (poorly) cover every possible option and it is a pretty easy search/replace when updating scripts.

Reply:
Where are the helper functions located? Am I blind?

Reply:
Never mind, found them...

Reply:
Also, whilst I prefer to have the underscores separator, I think I'm willing to accept its banishment to avoid something like this `f"{prefix}{group_designator}K{relay}" if "_" not in prefix else f"{prefix.strip('_')}{group_designator}_K{relay}" for relay in range(1, pin_count + 1)`
although, I do think that this would cover 95% of the way RMs have been defined to date.

Reply:
```python
def generate_pin_group(
    group_designator: int, *, pin_count: int = 16, prefix: str = "", sep: str = ""
) -> tuple[Pin, ...]:
    return tuple(
        f"{prefix}{group_designator}{sep}K{relay}" for relay in range(1, pin_count + 1)
    )

generate_pin_group(3, prefix="RM", sep="_")

('RM3_K1',
 'RM3_K2',
 'RM3_K3',
 'RM3_K4',
 'RM3_K5',
 'RM3_K6',
 'RM3_K7',
 'RM3_K8',
 'RM3_K9',
 'RM3_K10',
 'RM3_K11',
 'RM3_K12',
 'RM3_K13',
 'RM3_K14',
 'RM3_K15',
 'RM3_K16')
```

Maybe it's not too bad...

Reply:
That's waaaay better than my Frankenstein's monster... 🤦‍♂️

Reply:
I've relented, but I reserve the right to rant about how we use this inconsistently in the future... https://github.com/PyFixate/Fixate/pull/184/commits/16b81e3acf806f1cd8624f1cdb37e1043218bcb1